### PR TITLE
fix(ext/http): ensure aborted bodies throw

### LINKED ISSUE #20502: Aborted requests do not throw
If a connection hangs up before transmitting the body, `request.text()` should throw to indicate a failure.

```ts
Deno.test(
  { permissions: { net: true } },
  async function httpServerAbortedRequestBody() {
    const promise = deferred();
    const ac = new AbortController();
    const listeningPromise = deferred();

    const server = Deno.serve({
      handler: async (request) => {
        await assertRejects(async () => {
          await request.text();
        });
        promise.resolve();
        throw "Never reached";
      },
      port: servePort,
      signal: ac.signal,
      onListen: onListen(listeningPromise),
      onError: createOnErrorCb(ac),
    });

    await listeningPromise;
    const conn = await Deno.connect({ port: servePort });
    // Send GET request with a body + content-length.
    const encoder = new TextEncoder();
    const body =
      `POST / HTTP/1.1\r\nHost: 127.0.0.1:${servePort}\r\nContent-Length: 10\r\n\r\n12345`;
    const writeResult = await conn.write(encoder.encode(body));
    assertEquals(body.length, writeResult);
    conn.close();
    await promise;
    ac.abort();
    await server.finished;
  },
);
```

### PR DESCRIPTION
**Pull request by mmastrac**
Fixes #20502 -- ensure that Hyper errors make it through to JS.

### CODE REVIEW

File: ext/http/request_body.rs [hunk 1]
```diff
@@ -36,6 +36,9 @@ impl Stream for ReadFuture {
           }
         }
       }
+      std::task::Poll::Ready(Some(Err(e))) => {
+        return std::task::Poll::Ready(Some(Err(e.into())));
+      }
       std::task::Poll::Ready(None) => return std::task::Poll::Ready(None),
       _ => {}
```
Comment (dsherret) [inline ext/http/request_body.rs:43]: Bug caused by catch-all?

File: cli/tests/unit/fetch_test.ts [hunk 2]
```diff
@@ -1978,18 +1978,18 @@ Deno.test(
...
+//  assertEquals(actual, expected);
+//});
+//
+//Deno.test("Response with subarray TypedArray body", async () => {
+//  const body = new Uint8Array([1, 2, 3, 4, 5]).subarray(1);
+//  const req = new Response(body);
+//  const actual = new Uint8Array(await req.arrayBuffer());
+//  const expected = new Uint8Array([2, 3, 4, 5]);
+//  assertEquals(actual, expected);
+//});
```
Comment (bartlomieju) [inline cli/tests/unit/fetch_test.ts:1995]: Why disable this test?

File: ext/http/request_body.rs (hunk 1 above)
Comment (bartlomieju) [reply ext/http/request_body.rs:43]: I agree, let's add all arms explicitly

File: cli/tests/unit/fetch_test.ts (hunk 2 above)
Comment (mmastrac) [reply cli/tests/unit/fetch_test.ts:1995]: Ack. This was from another PR

File: ext/http/request_body.rs (hunk 1 above)
Comment (mmastrac) [reply ext/http/request_body.rs:43]: Good call. I will make it more explicit.

Comment (mmastrac) [reply ext/http/request_body.rs]: Rewrote this function as it had a chance to lose the waker as well

**Review by lucacasonato — APPROVED** [review]
LGTM

**Pull request merged**